pkg/asset/installconfig/gcp/validation.go

```diff
@@ -193,6 +194,13 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 			if len(ic.ControlPlane.Platform.GCP.Zones) > 0 {
 				zones = ic.ControlPlane.Platform.GCP.Zones
 			}
+			if ic.ControlPlane.Platform.GCP.DiskType != "" {
+				diskTypes := sets.NewString("pd-ssd", "pd-balanced", "hyperdisk-balanced")
```

Comment:
Let's use the non-deprecated set syntax
```suggestion
				diskTypes := sets.New("pd-ssd", "pd-balanced", "hyperdisk-balanced")
```